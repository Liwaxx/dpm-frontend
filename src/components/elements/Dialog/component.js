import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogContent from '@material-ui/core/DialogContent';
import PropTypes from 'prop-types';
import Text from '../Text';
import { IMAGES } from '../../../configs';

export default class Component extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
      scroll: 'paper',
      fullScreen: false,
    };
  }

  handleClickOpen = scroll => () => {
    this.setState({ 'open': true, scroll, fullScreen: true });
  };

  handleClose = () => {
    this.setState({ 'open': false });
  };

  render() {
    const { classes, children } = this.props;

    return (
      <div>
        <Button className={classes.bdetail}onClick={this.handleClickOpen('paper')}>Details</Button>
        <Dialog
          open={this.state.open}
          onClose={this.handleClose}
          scroll={this.state.scroll}
          className={classes.dialog}
          fullScreen={this.state.fullScreen}
        >
          <div className={classes.button}>
            <Text className={classes.title_squad}>Squad Backlog</Text>
            <button className={classes.close} onClick={this.handleClose}>
              <img src={IMAGES.CLOSE} className={classes.closeB} />
            </button>
          </div>
          <DialogContent className={classes.content}>{children}</DialogContent>
        </Dialog>
      </div>
    );
  }
}

Component.propTypes = {
  classes: PropTypes.object.isRequired,
  children: PropTypes.object
};