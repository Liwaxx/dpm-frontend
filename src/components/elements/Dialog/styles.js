const styles = ({
  dialog: {
    width: '90%',
    margin: 'auto',
    padding: 0,
  },
  title_squad: {
    textAlign: 'left',
    fontSize: 20,
    marginBottom: 5,
    width: '92%',
    float: 'left',
    outline: 'white',
    backgroundColor: 'white',
    border: 'white',
    paddingTop: 10,
    paddingLeft: 15
  },
  content: {
    padding: 0,
    paddingTop: 10,
  },
  bdetail:{
    float:'right',
    color:'#A8CEEB',
    bottom: 5,
    left:15
  },
  close: {
    width: '7%',
    float: 'right',
    paddingTop: 15,
    outline: 'white',
    backgroundColor: 'white',
    border: 'white'
  },
  closeB: {
    width: 15,
  },
});

export default styles;
