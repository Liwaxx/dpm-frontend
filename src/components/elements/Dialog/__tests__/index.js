import React from 'react';
import renderer from 'react-test-renderer';
import Dialog from '../index';

jest.mock('../../Text');
jest.mock('../../../../configs');

describe('Dialog', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Dialog />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});