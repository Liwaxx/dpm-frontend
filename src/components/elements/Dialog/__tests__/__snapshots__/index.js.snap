// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog renders correctly 1`] = `
<div>
  <button
    className="MuiButtonBase-root-33 MuiButton-root-7 MuiButton-text-9 MuiButton-flat-12 Component-bdetail-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-8"
    >
      Details
    </span>
    <span
      className="MuiTouchRipple-root-36"
    />
  </button>
</div>
`;
